vortex-array/src/data/viewed.rs

```diff
@@ -100,7 +101,16 @@ impl ViewedArrayData {
         self.encoding
             .accept(&ArrayData::from(self.clone()), &mut collector)
             .vortex_expect("Failed to get children");
-        collector.children
+        collector.children()
+    }
+
+    #[allow(dead_code)]
+    pub fn named_children(&self) -> Vec<(String, ArrayData)> {
+        let mut collector = NamedChildrenCollector::default();
```

Comment:
can we also add this to Owned and ArrayData similar to what we do for `children()` ?

Reply:
In theory we _only_ need to add it to ArrayData. We move the impl to Owned/Viewed if it gets big and we don't want to inline it, but it only needs to be pub(super)